pinot-transport/src/main/java/com/linkedin/pinot/routing/HelixExternalViewBasedRouting.java

```diff
@@ -130,76 +140,57 @@ public void markDataResourceOnline(String tableName, ExternalView externalView,
     }
 
     _routingTableLastKnownZkVersionMap.put(tableName, externalViewRecordVersion);
-    if (!_dataTableSet.contains(tableName)) {
-      LOGGER.info("Adding a new data table to broker : " + tableName);
-      _dataTableSet.add(tableName);
-    }
     RoutingTableBuilder routingTableBuilder;
-    TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableName);
-    if (tableType != null) {
-      switch (tableType) {
-        case REALTIME:
-          routingTableBuilder = _defaultRealtimeRoutingTableBuilder;
-          break;
-        case OFFLINE:
-          routingTableBuilder = _defaultOfflineRoutingTableBuilder;
-          break;
-        default:
-          routingTableBuilder = _defaultOfflineRoutingTableBuilder;
-          break;
-      }
+    CommonConstants.Helix.TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableName);
+
+    if (CommonConstants.Helix.TableType.REALTIME.equals(tableType)) {
+      routingTableBuilder = _realtimeHLCRoutingTableBuilder;
     } else {
-      routingTableBuilder = _defaultOfflineRoutingTableBuilder;
+      routingTableBuilder = _offlineRoutingTableBuilder;
     }
-    if (_routingTableBuilderMap.containsKey(tableName) && (_routingTableBuilderMap.get(tableName) != null)) {
-      routingTableBuilder = _routingTableBuilderMap.get(tableName);
-    }
-    LOGGER.info("Trying to compute routing table for table : " + tableName + ",by : " + routingTableBuilder);
+
+    LOGGER.info("Trying to compute routing table for table : " + tableName + " using " + routingTableBuilder);
+
     try {
       List<ServerToSegmentSetMap> serverToSegmentSetMap =
           routingTableBuilder.computeRoutingTableFromExternalView(tableName, externalView, instanceConfigList);
 
       _brokerRoutingTable.put(tableName, serverToSegmentSetMap);
+      if (CommonConstants.Helix.TableType.REALTIME.equals(tableType)) {
+        try {
+          List<ServerToSegmentSetMap> llcserverToSegmentSetMap = _realtimeLLCRoutingTableBuilder.computeRoutingTableFromExternalView(tableName, externalView, instanceConfigList);
+
+          _llcBrokerRoutingTable.put(tableName, llcserverToSegmentSetMap);
+        } catch (Exception e) {
+          LOGGER.error("Failed to compute LLC routing table for {}. Ignoring", tableName, e);
+        }
+      }
     } catch (Exception e) {
       LOGGER.error("Failed to compute/update the routing table" + e.getCause(), e);
     }
+
     try {
       LOGGER.info("Trying to compute time boundary service for table : " + tableName);
       _timeBoundaryService.updateTimeBoundaryService(externalView);
     } catch (Exception e) {
       LOGGER.error("Failed to update the TimeBoundaryService : " + e.getCause(), e);
```

Comment:
I realize this is existing code, but since you are already modifying the file, it would be good to cleanup the usage of '+' and replace with '{}' in LOGGER.info/error calls.